modules/nf-core/kraken2/add/main.nf

```diff
@@ -0,0 +1,38 @@
+process KRAKEN2_ADD {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' :
+        'biocontainers/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+    path taxonomy_names
+    path taxonomy_nodes
+    path accession2taxid
```

Comment:
IF we stick with the standalone `ADD` module - would this still be necessary? I'm assuming yes, but we need to be careful how this works as these would have to be staged into an existing `$prefix/taxonomy` directory, and make sure this doesn't get overwritten on every `add`...

Which also leads to the question, what do we do if we want to add multiple FASTAs in one (so `path(fasta)` is a list)...? The examples in the kraken2 manual suggest using for loops or `find/xargs`  combination

Reply:
> IF we stick with the standalone `ADD` module - would this still be necessary? I'm assuming yes, but we need to be careful how this works as these would have to be staged into an existing `$prefix/taxonomy` directory, and make sure this doesn't get overwritten on every `add`...
> 
> Which also leads to the question, what do we do if we want to add multiple FASTAs in one (so `path(fasta)` is a list)...? The examples in the kraken2 manual suggest using for loops or `find/xargs` combination

That's correct, I forgot to put all those files in the taxonomy directory (db/taxonomy) as expected by kraken2.